app/src/main/java/com/rkm/tasky/feature/loginscreen/screen/LoginScreen.kt

```diff
@@ -0,0 +1,229 @@
+import com.rkm.tasky.ui.component.ValidationTextField
+import com.rkm.tasky.ui.theme.LoginSignUpTextColor
+
+@Composable
+fun LoginScreenRoot(
+    navHostController: NavHostController,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+
+    val email by viewModel.emailState.collectAsStateWithLifecycle()
+    val isValidEmail by viewModel.isValidEmailState.collectAsStateWithLifecycle()
+    val password by viewModel.passwordState.collectAsStateWithLifecycle()
+    val showPassword by viewModel.showPasswordState.collectAsStateWithLifecycle()
+
+    LoginScreen(
+        email = email,
+        password = password,
+        isValidEmail = isValidEmail,
+        showPassword = showPassword,
+        onEmailChange = viewModel::updateEmail,
+        onPasswordChange = viewModel::updatePassword,
+        onLoginButtonClick = { /*TODO*/ },
+        modifier = Modifier
+    )
+
+}
+
+@Composable
+fun LoginScreen(
+    email: String,
+    password: String,
+    isValidEmail: Boolean,
+    showPassword: Boolean,
+    onEmailChange: (String) -> Unit,
+    onPasswordChange: (String) -> Unit,
+    onLoginButtonClick: () -> Unit,
+    modifier: Modifier
+) {
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(color = Color.Black),
+    ) {
+        Column(
+            modifier = modifier
+                .fillMaxWidth()
+                .fillMaxHeight(0.2f),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center,
+        ) {
+
+            Text(
+                modifier = Modifier
+                    .padding(8.dp),
+                text = stringResource(R.string.login_screen_greeting),
+                style = TextStyle(
+                    color = Color.White,
+                    fontSize = MaterialTheme.typography.headlineLarge.fontSize,
+                    fontWeight = FontWeight.Bold
+                ),
+            )
+        }
+
+        Column(
+            modifier = modifier
+                .fillMaxSize()
+                .clip(RoundedCornerShape(topStart = 30.dp, topEnd = 30.dp))
+                .background(Color.White),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            LoginBodyScreen(
+                modifier = modifier,
+                email = email,
+                password = password,
+                isValidEmail = isValidEmail,
+                showPassword = showPassword,
+                onEmailChange = onEmailChange,
+                onPasswordChange = onPasswordChange,
+                onLoginButtonClick = onLoginButtonClick
+            )
+
+            Spacer(modifier = Modifier
+                .fillMaxWidth()
+                .fillMaxHeight(0.7f)
+            )
+
+            SignUpText(
+                modifier = modifier,
+                onSignUpClick = {}
+            )
+        }
+
+    }
+
+}
+
+@Composable
+fun LoginBodyScreen(
```

Comment:
These composables seem specific to this screen, its best to indicate this with the `private` keyword to prevent its usage outside of this file